### PR TITLE
Support route tables with propagatingVgwSet.

### LINKED ISSUE #3049: Incorrect boto.vpc.routetable.RouteTable objects with propagated routes
When investigating a boto.vpc.routetable.RouteTable object I noticed the following behaviour:

If the route table has route propagation enabled, we see 'half baked' route table objects being created, containing just tags. The behaviour is:

If no route propagation is enabled; create a correct route table object.
If route propagation _is_ enabled; create a route object with tags set to an empty dict ({}). Next, for each tag in the tagSet; create a route object containing no routes, and 'key' set to the key of the fist tag, 'value', set to the value of the first tag.

Ipython example:

Normal Route object:

```
In [7]: vars(all_route_tables[1].routes[2])
Out[7]:
{'destination_cidr_block': u'1.2.3.4/16',
 'gateway_id': u'vgw-123',
 'instance_id': None,
 'interface_id': None,
 'state': u'active',
 'vpc_peering_connection_id': None}
```

Route object with route propagation, notice tags are empty:

```
In [9]: vars(all_route_tables[1])
Out[9]:
{'associations': [RouteAssociation:rtbassoc-123],
 'connection': VPCConnection:ec2.us-east-1.amazonaws.com,
 'gatewayId': u'vgw-123',
 'id': u'rtb-123',
 'item': u'\n                ',
 'region': RegionInfo:us-east-1,
 'routes': [Route:1.2.3.4/17,
  Route:0.0.0.0/0,
  Route:1.2.3.5/16,],
 'tags': {},
 'vpc_id': u'vpc-123'}
```

The next Route object contains the tags of the previous one:

```
{'associations': [],
 'connection': VPCConnection:ec2.us-east-1.amazonaws.com,
 'id': None,
 'item': u'\n                ',
 'key': u'Foo',
 'region': RegionInfo:us-east-1,
 'routes': [],
 'tags': {},
 'value': u'Bar',
 'vpc_id': None}
```

And so on.

PR #3046 fixes this issue, and adds tests for it. PR #3047 contains the fix and adds a test for the route origin.

### PR DESCRIPTION
**Pull request by ojongerius**
Fixes #3049

### CODE REVIEW

**Comment by mtdowling** [comment]
Taking a look

File: boto/vpc/routetable.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def __init__(self, connection=None):
         self.id = None
         self.vpc_id = None
         self.routes = []
+        self.propagatingvgws = []
```
Comment (mtdowling) [inline boto/vpc/routetable.py]: Any reason this is not `propagating_vgws` instead of all one word?

**Comment by mtdowling** [comment]
I don't see any tests added for actually asserting that `propagating_vgws` is actually populated. Can you make sure this is tested please?

Can you also rebase against the latest develop?

**Comment by ojongerius** [comment]
Closing in favour of #3047.

**Pull request closed without merging**